lms/djangoapps/verify_student/signals.py

```diff
@@ -14,11 +14,11 @@ def _listen_for_course_publish(sender, course_key, **kwargs):  # pylint: disable
     Catches the signal that a course has been published in Studio and
     sets the verification deadline date to a default.
     """
-    try:
-        deadline = VerificationDeadline.objects.get(course_key=course_key)
-        if deadline and not deadline.deadline_is_explicit:
-            course = modulestore().get_course(course_key)
-            if course and deadline.deadline != course.end:
+    course = modulestore().get_course(course_key)
+    if course:
+        try:
+            deadline = VerificationDeadline.objects.get(course_key=course_key)
+            if not deadline.deadline_is_explicit and deadline.deadline != course.end:
                 VerificationDeadline.set_deadline(course_key, course.end)
-    except ObjectDoesNotExist:
-        pass
+        except ObjectDoesNotExist:
+            VerificationDeadline.set_deadline(course_key, course.end)
```

Comment:
Is there a reason that this logic needs to be reworked? It seems like the end result is the same.